### PR TITLE
SW-2159 Update Dialog Skrim

### PR DESCRIPTION
**Pull request by charlie-tf**
- skrim off by default
- should be on for destructive actions only
- remove blur
- change background to gradient
image
image

### CODE REVIEW

File: src/components/DialogBox/styles.scss [hunk 1]
```diff
@@ -17,6 +15,17 @@
   z-index: 1300;
   inset: 0px;
 
+  &--skrim {
+    background: linear-gradient(180deg, rgba($tw-clr-base-gray-025, $tw-opcty-base-060), rgba(#EFF5EF, $tw-opcty-base-060));
```
Comment (charlie-tf) [inline src/components/DialogBox/styles.scss:19]: Asked for a token for `#EFF5EF`, we didn't think we'd need it at first, but it would be helpful here.

Comment (karthikbtf) [reply src/components/DialogBox/styles.scss:19]: should we be using semantic tokens here (for the ones with tokens)?
gives designers more control

Comment (charlie-tf) [reply src/components/DialogBox/styles.scss:19]: was told that base colors should be the way to go for this background gradient

Comment (karthikbtf) [reply src/components/DialogBox/styles.scss:19]: fair enought

**Review by karthikbtf — APPROVED** [review]
👍

File: src/components/DialogBox/styles.scss [hunk 2]
```diff
@@ -17,6 +15,18 @@
   z-index: 1300;
   inset: 0px;
 
+  &--skrim {
+    background-color: rgba($tw-clr-base-gray-025, $tw-opcty-base-060);
+    background-image: linear-gradient(180deg, rgba($tw-clr-base-green-050, $tw-opcty-base-000), rgba($tw-clr-base-green-050, 0.24));
```
Comment (charlie-tf) [inline src/components/DialogBox/styles.scss:20]: I was told by @hannah-cai that this was how I should construct the background gradient. Namely, a solid `tw-clr-base-gray-025` overlaid by a `tw-clr-base-green-050` gradient from transparent to 40% (for the skrim, the opacities were multiplied by 0.6 per the spec).

Comment (karthikbtf) [reply src/components/DialogBox/styles.scss:20]: 👍 
@charlie-tf can we loop @chudo76 in here... using these base tokens is no different from using hardcoded colors, etc.

File: src/components/DialogBox/DialogBox.tsx [hunk 3]
```diff
@@ -15,18 +15,19 @@ export interface Props {
   onClose?: () => void;
   open: boolean;
   scrolled?: boolean;
+  skrim?: boolean;
```
Comment (karthikbtf) [inline src/components/DialogBox/DialogBox.tsx:18]: oops, missed asking to add this to the stories

Comment (karthikbtf) [reply src/components/DialogBox/DialogBox.tsx:18]: ugh, my bad - it's there. just tried it. all good 👍

**Review by karthikbtf — APPROVED** [review]
👍 
my comment re: base tokens.. we can follow up on that later.
would be great to use semantic tokens as much as possible.

File: src/components/DialogBox/styles.scss (hunk 2 above)
Comment (hannah-cai) [reply src/components/DialogBox/styles.scss:20]: @chudo76 and I have agreed that not every use of color needs to have a semantic token, we will be using base colors occasionally for things that are more or less one-offs, and which don't have a semantic meaning

Comment (karthikbtf) [reply src/components/DialogBox/styles.scss:20]: Thanks for clarifying @hannah-cai , sounds good 👍

**Comment by hannah-cai** [comment]
> * skrim off by default
> * should be on for destructive actions only
> * remove blur
> * change background to gradient
> 
> image
> 
> image

unrelated to this branch but it's been bothering me for a while that dialog headers are off-center. is that already in the pipeline to be fixed?

**Comment by charlie-tf** [comment]
>unrelated to this branch but it's been bothering me for a while that dialog headers are off-center. is that already in the pipeline to be fixed?

@hannah-cai I can take a look at it on this branch. I'm not aware of whether it's in the pipeline

**Comment by charlie-tf** [comment]
centered!
image

**Comment by karthikbtf** [comment]
> centered! image

Nice, that's one way to do it, balance it out with a symmetric spacer. 
I was expecting to see a text-align:center and icon floated right or absolutely positioned on the right.
The spacer approach works better with flex for sure. 
All good 🎉

**Pull request merged**